### ISSUE TITLE
Feature request and more

### ISSUE THREAD

**Issue by babakfp** [sass/sass#3066]
## 1. `.sass` syntax issue
in a `.sass` file you need to write all function arguments in one single line.
```sass
@function($a: a, $b: b, $c: c)
```
### Solution
I request to remove this limitation for the function arguments and other stuff like `@mixin()`:
```scss
@function(
  $a: a,
  $b: b,
  $c: c
)
```

## 2. How we create a list type value
```sass
$a: 1 2 3
@debug type-of($a) // list
$b: 1, 2, 3
@debug type-of($b) // list

$c: [1 2 3]
@debug type-of($c) // list
$d: [1, 2, 3]
@debug type-of($d) // list

$e: (1 2 3)
@debug type-of($e) // list
$f: (1, 2, 3)
@debug type-of($f) // list

$g: ((a: a))
@debug type-of($g) // map :|
$h: ((a: a),)
@debug type-of($h) // list :|
$i: (())
@debug type-of($i) // list
$j: ((),)
@debug type-of($j) // list
```
What the heck is this!

### Solution
The only way to create a list type value must be like the `$d` example:
```sass
$d: [1, 2, 3]
@debug type-of($d) // list
```
Also, we shouldn't be able to create maps like this: (we must have a string value as the key of the value)
```sass
@use "sass:map"

$k: ((): value)
@debug type-of($k) // map

@debug map.get($k, ()) // value
```

## 3. Importing core stuff
I don't like to import code functions like (`sass:map`, `sass:list`), etc.

## 4. Syntax of how we interact with maps
This is how you use maps right now:
```scss
@use "sass:map";
map.get($my-map, my-key)
```

### Solution
```SCSS
$my-map.my-key
```

## 4. `type-of($something)` syntax
It's better to be like: 
```scss
typeof $something
@if typeof $something {}
```

## 5. Chaining `@include`
```sass
.test
  @include a-mixin another-mixin third-mixin(1px)
```

## Stylus
1. [Binary Operators - Subscript](https://stylus-lang.com/docs/operators.html#subscript-). It's better to start from 1, not 0.
2. [Binary Operators - Range](https://stylus-lang.com/docs/operators.html#range--). I'm not agree with `...`.

**Comment by Awjin** [comment]
1. This is a duplicate of https://github.com/sass/sass/issues/216.
2. This syntax exists due to backwards compatibility with CSS, and cannot be changed as you described.
3. The module system was designed this way to enable more explicit dependencies.
4. It does not make sense to change the syntax of this call (thereby making it inconsistent with the rest of Sass) in order to mirror JS syntax.
5. Each include must be declared separately, consistent with other at-rules like `@use`.
6. Either of these changes would be enormous breaking changes to the entire Sass ecosystem.

**Comment by babakfp** [comment]
Why you immediately closed the issue? Is there a party you need to go to? Why did you rush?

1. I bet $1 that this problem is not going to be solved at all.
2. which "backward compatibility"? I'm suggesting just a single syntax `[a, b]` because there are many problems caused by this.
3. Why not design it and gave it global access? It does not make sense. programing languages have 100000 features without needing to import something. even for using a math division, we need to import something. What the hell!
4. Are we talking about consistency? I can show you a hell of inconsistency in SASS.
5. Why? It is just dumb and not friendly.
6. I hear you, it's hard. Let's just don't do it.

I guess we are gonna wait for a new thing like sass. Already half of the world not uses sass so it's in a progress to taking the sass to the bed forever. I always discussed about SASS with other people and said good things about it because it's awesome. but now when I see this kinda stuff, I'm like whatever, it doesn't even worth it. I can just write not-so-beautiful CSS and don't have this kinda sad stuff.

**Comment by nex3** [comment]
@babakfp Please recall the [Community Guidelines](https://sass-lang.com/community-guidelines): disagreement is no excuse for poor manners. Awjin addressed all of your points, and there's no action the Sass team plans to take here, so he closed the issue.  He didn't lock it or in any way prevent further discussion. This is a normal part of issue triage for any open-source project.

> 1. I bet $1 that this problem is not going to be solved at all.

Sass is an open source project with *many* important outstanding feature requests and much less engineer-power than necessary to complete all of them. There are going to be valid, useful features that don't get implemented for years—that's just the way engineering works. Complaining about it does nothing but make you look whiny and immature.

If you want this to happen, you can either go through the new feature process to add it yourself or start raising money to hire someone to implement it.

> 2. which "backward compatibility"? I'm suggesting just a single syntax `[a, b]` because there are many problems caused by this.

CSS frequently writes lists of values without any kind of bracket, as in `border: 1px solid black` and `font-family: Segoe UI, Helvetica, Arial, sans-serif, Apple Color Emoji, Segoe UI Emoji`. Sass has a strong commitment to CSS compatibility, so we must support these forms of lists as well.

> 3. Why not design it and gave it global access? It does not make sense. programing languages have 100000 features without needing to import something. even for using a math division, we need to import something. What the hell!

Global function names can and do cause conflicts when CSS introduces functions with the same names. When CSS introduced `min()` and `max()` functions, we had to go through a [long and painful process](https://github.com/sass/sass/issues/2378) of disambiguating Sass's built-in functions from CSS's. If the CSS syntax had been slightly different, this disambiguation might not have been possible at all. Providing explicit namespaces for Sass functions is crucial for avoiding similar compatibility issues in the future.

> 4. Are we talking about consistency? I can show you a hell of inconsistency in SASS.

Sass has a strong policy of introducing as little syntax as possible on top of CSS's baseline syntax. Once again, CSS can and has adopted syntax that Sass was using for something different, and caused extremely painful compatibility issues along the way. Supporting `$map.key` or `$map[key]` would add a huge amount of risk for a very small amount of value over writing `map.get($map, key)`.

> 5. Why? It is just dumb and not friendly.

The more different ways we have of doing the same thing, the more the door opens for inconsistency and confusion from one user to another. Maximal terseness is never our goal. We want Sass to be clearly and immediately understandable even by users who are not very experienced with it, and we want Sass knowledge to be easily transferred from one project to another. Saving a few `@include`s provides very little value relative to the cost of having multiple ways to write multiple mixin invocations.

> 6. I hear you, it's hard. Let's just don't do it.

See above about why we don't add new syntax unless we absolutely have to.

**Comment by babakfp** [comment]
@nex3 
Hello
This is a product and as a user, I have the right to complain, discous about new features, etc.
Answering issues isn't about just reading and closing them. If you guys don't want to answer the issues, simply just close the whole section and make it inaccessible. If you are going to take your time to write an answer, write a good one.
Closing the issue right away is just disrespectful (not in all scenarios).
If an issue gets closed, it means that it wasn't important/needed. It means that the maintainers don't care about it. It means the users aren't going to be able to find it easily.

### about 1.
If there is a lack of engineers to wrap up the requested/planned features, then you will need to do something about it. for example, writing good documentation to tech to the interested developers on how sass works, how they can implement new features, etc. Not doing anything means you don't care enough.
the Users are just a user. You shouldn't expect them to do a lot or anything. As a user, I just can do some stuff to improve stuff, report bugs, etc.

If someone complains about something, it shows that there is a problem with that thing. Also, I didn't complain, I just simply report and said my thoughts.

If the sass team lost their interest in sass, I recommend them to publicly announce it. Because there may be someone else interested enough to take over development. Also, it's a necessity to notify the users about it because they may don't want to use sass anymore and look for a new tool. Maybe by doing this, another tool going to be popular and better than sass in the feature.

### about 2.
If CSS uses example `$a` and `$b` then let's just have those 2 syntaxes. CSS doesn't use example `$c`, `$d`, `$e`, `$f`, `$h`, `$i`, `$g`.
There is an issue as you see in the example `$g` and `$h`. After that, you can see that there is another issue `$k: ((): value)`.

### about 3.
Hmm, I didn't know about sass `min()` and `max()` functions.
We also need to import the `sass:something`. The solution was something that no developer likes.

#### Solution

Add a prefix like:
```sass
// before:
@use "sass:map"
map.deep-merge($map-one, $map-two)

// after:
$deep-merge($map-one, $map-two)
// or
map.deep-merge($map-one, $map-two)
```

### about 4.
If someone uses sass, it means that they work with HTML and CSS. It's 99% possible that they going to learn Javascript. If so, they can be familiar with the syntax `$map[key]` and it can help them in the JS journey. It can be a little positive point of learning sass for them.
#### If we don't have the requested syntax
- We are going to write 26 characters more than you need.
- We need to repeat the same in every single file that you use maps.
- It's going to make the code look messier and unreadable.
I'm giving too much for not having the proper syntax.

### about 5.
How we use the `@extend` rule?
```scss
.a {
  a: a;
}
.b {
  b: b;
}
.c {
  @extend .a, .b;
  c: c;
}
// It works
```
Why we can't do the same? We already have it with the `@extend` rule. How doing it this way makes it less understandable? What about the cost of being able to create a list in multiple(9) ways (two of them as you said are for the CSS syntax)?

### not related to this issue but I'm getting an error with the .sass syntax
```sass
// a file names .sass
.a
  a: a
.b
  b: b
.c
  @extend .a, .b
  c: c
```
The error:
```terminal
PS C:\Users\babak\Desktop\New folder> sass .sass:.css      
Error: expected "{".
  ╷
7 │   c: c
  │       ^
  ╵
  .sass 7:7  root stylesheet
PS C:\Users\babak\Desktop\New folder> 
```
Why it expects a `{`! Is this is a wrong error message or a bug🤔?

### about 6.
Why not adding a needed feature is better than adding it? It doesn't sound right.

**Comment by nex3** [comment]
You have absolutely no right to demand time and labor from people who are giving you something of value for free. We *choose* to maintain an issue tracker and engage with users because we value and respect our userbase, but that is entirely contingent on users valuing and respecting us in turn. If you're going to be snarky or rude about the way we triage our issues, you're not a user we're interested in engaging with. To that point, I'm going to lock this issue now.

To your point about the Sass team "losing their interest", Sass is more heavily staffed now than it ever has been in the 15 years it exists. Throwing more engineers at a project doesn't magically change the prioritization of existing issues, though, and no matter how much time we put into it there will *always* be features that fall by the wayside. That's just the reality of any software development.
